jetstream/tests/test_cli.py

```diff
@@ -486,29 +480,35 @@ def test_simple_workflow(self, cli_experiments, monkeypatch, docker_images):
         config = spec.resolve(experiment, ConfigLoader.configs)
         mock_artifact_client = Mock()
         mock_artifact_client.list_docker_images.return_value = docker_images
-        monkeypatch.setattr(BigQueryClient, "experiment_table_first_updated", lambda _, slug: None)
         monkeypatch.setattr(ArtifactManager, "client", property(lambda _: mock_artifact_client))
 
         with mock.patch("jetstream.cli.submit_workflow") as submit_workflow_mock:
-            strategy = cli.ArgoExecutorStrategy(
-                project_id="spam",
-                dataset_id="eggs",
-                bucket="bucket",
-                zone="zone",
-                cluster_id="cluster_id",
-                monitor_status=False,
-                cluster_ip=None,
-                cluster_cert=None,
-                experiment_getter=lambda: cli_experiments,
-                analysis_periods=[
-                    AnalysisPeriod.DAY,
-                    AnalysisPeriod.WEEK,
-                    AnalysisPeriod.DAYS_28,
-                    AnalysisPeriod.OVERALL,
-                ],
-            )
-            run_date = dt.datetime(2020, 10, 31, tzinfo=UTC)
-            strategy.execute([(config, run_date)])
+            with mock.patch("jetstream.artifacts.BigQueryClient") as bq_client:
+                bigquery_mock_client = MagicMock()
+                bigquery_mock_client.experiment_table_first_updated.return_value = dt.datetime(
+                    2023, 4, 1, tzinfo=UTC
+                )
+                bq_client.return_value = bigquery_mock_client
```

Comment:
Why can't this use the mocked client from conftest?

Reply:
This one is specifically mocking the `BigQueryClient` instance used in `ArtifactManager`. The other fixture is mocking the instances used in the CLI